### DISCUSSION TITLE
Unexpected behavior of user-defined boundary condition

### DISCUSSION THREAD

**Discussion by kfxw** [mdolab/adflow#340] [Q&A] [answered]
Hello,

I'm using ADFlow to run a simple simulation using a mesh built on a sphere with pyHyp. I define the boundary conditions with various ways as shown below. However, the solutions keep giving me constant pressure, density and velocity values no matter what values I assigned to the altitude, Reynolds number, temperature... This makes me very confused.

I contain a minimal case to reproduce the issue in the attached file. Please feel free to have a try if this problem also confuses you. Thanks for any comments.

[minimal_reproduction.zip](https://github.com/mdolab/adflow/files/14088183/minimal_reproduction.zip)

```python
...
CFDSolver = ADFLOW(options=aeroOptions, comm=comm)

# define aero problem
# air at 20℃ -> Kinematic Viscosity m2/s: 1.5111E-5

# v = 3 m/s = 10.8 km/h, Re = 198531
#ap = AeroProblem(
#    name="test_solver_output", \
#    V=3.0, rho=1.225, T=293.15, evalFuncs=["lift", "drag"], \
#    areaRef = 0.7854, chordRef = 1
#)
# PS: it is the same if using a different way to initialize AP
ap = AeroProblem(
    name="test_solver_output", \
    V=3.0, reynolds=198531.0, reynoldsLength=1.0, T=293.15, P=100982, rho=1.2, evalFuncs=["lift", "drag"], \
    areaRef = 0.7854, chordRef = 1
)

# or
#ap = AeroProblem(
#    name="test_solver_output", \
#    mach=0.0087, reynolds=198531.0, reynoldsLength=1.0, T=293.15, evalFuncs=["lift", "drag"], \
#    areaRef = 0.7854, chordRef = 1
#)

# or
#ap = AeroProblem(
#    name="test_solver_output", \
#    mach=0.0087, altitude=100, evalFuncs=["lift", "drag"], \
#    areaRef = 0.7854, chordRef = 1
#)
# simulation
funcs = {}
CFDSolver(ap)

# but the solution always gives Pressure=1, Density=1, Velocity=0.01
```

**Comment by marcomangano** [comment]
Hello,
First, it looks like you are over-defining the problem in the uncommented case - double check [this docstring](https://mdolab-baseclasses.readthedocs-hosted.com/en/latest/pyAero_problem.html#baseclasses.AeroProblem) for the available combinations of thermodynamic parameters.

Secondly, how are you checking the solution values? Are you looking at the output files? Or are you extracting them from the `AeroProblem` instance? For security reasons we do not handle `zip` files from external users.

**Reply by kfxw** [reply]
Hello,
Thanks for the reply. If you don't mind, please download my script and the mesh using the link below, since github does not allow uploading .py and .cgns files.
https://filesender.renater.fr/?s=download&token=f6f33948-c8a2-4133-aa0c-ea16a793f4b1

Indeed the uncommented codes is over-defining, but the other commented ones gave me the same result.

As for checking the solution, I used ParaView to visualize the solution's velocity and pressure values in the far field. They were very far away from my definitions in the AeroProblem. For example, the velocity keeps being 0.1m/s when I specified V=3 in the code. Same to the pressure and density. 

Since I assume it would be the initialization issue, so I stop the simulation at the first iteration in my reproduction script, but running it to convergence would give a similar result.

Thanks for your time.

**Reply by marcomangano** [reply, marked as answer]
Ok, I understand the issue you are having. What you are seeing is actually nominal ADflow behavior.

The solver uses non-dimensional thermodynamic variables throughout the code (you can take a peek at [this initialization Fortran routine](https://github.com/mdolab/adflow/blob/aff9fcb12f976967cca4205952b424223794bfbc/src/initFlow/initializeFlow.F90#L57-L109) if you are curious) and does not convert back to the dimensional value when writing the output file.

We generally look at normalized pressure and velocity contours when we check and discuss results. If you need dimensional values, you can just convert them back in Paraview or Tecplot, multiplying the pressure and density using your input values and / or the full thermodynamic state from the `AeroProblem` function I mentioned in my first comment. 
Note that the velocity used for normalization (`uInf` [here](https://github.com/mdolab/adflow/blob/aff9fcb12f976967cca4205952b424223794bfbc/src/initFlow/initializeFlow.F90#L83)) is based on the freestream Mach number and gamma (`pInf` and `rhoInf` are already normalized).
Hope this is what you need to move on with your work.

**Reply by kfxw** [reply]
Thanks very much. It explains everything for me!
